p2p/src/peer_manager/mod.rs

```diff
@@ -115,50 +129,36 @@ where
         !self.peerdb.is_id_banned(peer_id)
     }
 
-    /// Verify protocol compatibility
+    /// Verifies the protocols compatibility.
     ///
-    /// Make sure that remote peer supports the same versions of the protocols that we do
-    /// and that they support the mandatory protocols which for now are configured to be:
+    /// Checks that the given set of protocols contains the following protocols with the exact
+    /// versions:
     ///
-    /// - `/meshsub/1.1.0`
-    /// - `/meshsub/1.0.0`
-    /// - `/ipfs/ping/1.0.0`
-    /// - `/ipfs/id/1.0.0`
-    /// - `/ipfs/id/push/1.0.0`
-    /// - `/mintlayer/sync/0.1.0`
+    /// - Pubsub 1.1.0
+    /// - Ping 1.0.0
+    /// - Sync 0.1.0
     ///
-    /// If any of the procols are missing or if any of them have a different version,
+    /// If any of the protocols are missing or if any of them have a different version,
     /// the validation fails and connection must be closed.
     ///
     /// Either peer may support additional protocols which are not known to the other
     /// peer and that is totally fine. As long as the aforementioned protocols with
     /// matching versions are found, the protocol set has been validated successfully.
-    // TODO: create generic versions of the protocols when mock interface is supported again
-    // TODO: convert `protocols` to a hashset
-    // TODO: define better protocol id type
-    fn validate_supported_protocols(&self, protocols: &[T::ProtocolId]) -> bool {
-        const REQUIRED: &[&str] = &[
-            "/meshsub/1.1.0",
-            "/meshsub/1.0.0",
-            "/ipfs/ping/1.0.0",
-            "/ipfs/id/1.0.0",
-            "/ipfs/id/push/1.0.0",
-            "/mintlayer/sync/0.1.0",
+    fn validate_supported_protocols(&self, protocols: &HashSet<Protocol>) -> bool {
+        const REQUIRED: &[Protocol] = &[
+            Protocol::new(ProtocolType::Pubsub, SemVer::new(1, 0, 0)),
+            Protocol::new(ProtocolType::Pubsub, SemVer::new(1, 1, 0)),
```

Comment:
It is a bit of an unfortunate detail of libp2p that they advertise both `/meshsub/1.1.0` and `/meshsub/1.0.0` but it is what it is. I think the correct way, as long as they keep advertising both, is to either parse the combination of these two into `Protocol::new(ProtocolType::Pubsub, SemVer::new(1, 1, 0))` (so there's only one PubSub protocol reported to `PeerManager`) or the check for required protocols is adjusted to verify that there exists at least one PubSub protocol that is compatible with our implementation.

The reason is that the mock interface won't have two PubSub implementations for now at least so requiring two versions of the protocol from `MockService` doesn't make sense.